### DISCUSSION TITLE
WeAct STM32G431CBU6

### DISCUSSION THREAD

**Discussion by ITstreet1** [stm32duino/Arduino_Core_STM32#2286] [Q&A] [answered]
Hi,
I have a board from the title
https://www.aliexpress.com/item/1005005303669884.html

Tried to upload a sketch both over the UART and SWD, but no luck.

When using the SWD I get the message that the board is not detected.

When using the UART, I get the sketch uploaded, but a simple Blink does not work.

Of course, I use a proper upload method, board selection, etc.
Any suggestion?

PS:
Please someone to add in the title something like upload problem, or so. I forgot. Sry

**Comment by fpistm** [comment]
Hi,
which target you select ?
Core version?
...

**Comment by ITstreet1** [comment]
HI,
Board generic STM32G4
Board part number generic G431CBUx
The rest is the default
Upload method STM32Cube programmer (serial)

Tried Blink Sketch for PC6 (built-in LED), PC4, etc. Nothing. While using UART, I upload the sketch, but nothing on pins. I restart the board after uploading. Nothing

PS:
UART RX - PA9
UART TX - PA10
Edit2: Not PC but PA.

**Reply by fpistm** [reply]
Hi @ITstreet1 
This variant was added by a contributor  to add the B-G431B-ESC1(including the generic).
I do not have  board with STM32G431CB to test.
Anyway, if the generic does not work it's probably a system core clock config or your board have an issue.
If the clock config failed then the board reach the error handler. You could try to debug to see where you stuck.

**Reply by ITstreet1** [reply]
How to do that? What to use?

**Reply by fpistm** [reply]
Arduino IDE 2 allows to debug. Select optimize for debug in the menu. Build, upload the launch debug using the third icon. You need a ST Link.

**Reply by ITstreet1** [reply]
OK.
I tried STM32CubeProgrammer connected with the ST-Link V2. It throws:

>   11:59:03 : ST-LINK SN  : 48FF6F064982565039210787
>   11:59:03 : ST-LINK FW  : V2J39S7
>   11:59:03 : Board       : --
>   11:59:03 : Voltage     : 3.25V
>   11:59:03 : Error: No STM32 target found!

The same, but with UART:

>   12:01:42 : RTS low
>   12:01:42 : DTR low
>   12:01:42 : Serial Port COM20 is successfully opened.
>   12:01:42 : Port configuration: parity = even, baudrate = 115200, data-bit = 8,                     stop-bit = 1.0, flow-control = off
>   12:01:44 : Timeout error occured while waiting for acknowledgement.
>   12:01:44 : Timeout error occured while waiting for acknowledgement.
>   12:01:44 : Error: Activating device: KO. Please, verify the boot mode configuration and check the serial port configuration. Reset your device then try again... 

I reset the board while pressing the BOOT button. But, it looks like it didn't enter bootloader mode. 

Any suggestion?

**Reply by ITstreet1** [reply]
Tried to upload another sketch to the board over the UART. Now it won't.

**Reply by fpistm** [reply]
If the stm32cube programmer can't connect there is an issue with your board. hard to help.

**Reply by ITstreet1** [reply]
Could it be I bricked it somehow?

I did get an upload success over the UART but with no response on the pins. Now I can not upload at all.

**Reply by ITstreet1** [reply]
Just tried Arduino IDE 2.0. There is a message that I don't have a selected programmer. But where to select it? There is no option.

**Reply by fpistm** [reply]
You have to use the 2.3.2 version of the IDE. 2.3.0 and 2.3.1 broke the debug feature.
Anyway, if you can't connect with CubeProgrammer the debug would probably not work. While you are not able to connect using CubeProgrammer only you will not be able to use Arduino.

**Reply by ITstreet1** [reply]
You are right. I can't... Any possibility I bricked it somehow? All I did was try to upload a simple Blink sketch.

**Reply by fpistm** [reply]
I understood, but if you simply can't connect within STM32CubeProgrammer this means it exists an issue not link to Arduino.

**Reply by fpistm** [reply, marked as answer]
I guess this is this board:
https://github.com/WeActStudio/WeActStudio.STM32G431CoreBoard

Do you connect the NRST signal of the STLink ?

**Reply by ITstreet1** [reply]
This is one of the most stupid mistakes one can make. And I did.

Thanks

**Reply by fpistm** [reply]
So you can connect now?

**Reply by ITstreet1** [reply]
Yes, it can.
